include/ignition/gazebo/components/Component.hh

```diff
@@ -282,7 +282,7 @@ namespace components
 
     /// \brief Clone the component.
     /// \return A pointer to the component.
-    public: virtual std::unique_ptr<BaseComponent> Clone() = 0;
+    public: virtual std::unique_ptr<BaseComponent> Clone() const = 0;
```

Comment:
Good call. I think this could be broken into a separate PR.

Reply:
Included in #1300